src/rgw/rgw_common.cc

```diff
@@ -987,7 +987,7 @@ bool verify_object_permission(struct req_state * const s,
   }
 
   bool ret = object_acl->verify_permission(*s->auth_identity, s->perm_mask, perm);
-  if (ret) {
+  if (ret && s->dialect.compare("swift") != 0) {
```

Comment:
I would propose to use here `s->prot_flags` and  the `RGW_REST_SWIFT` flag. I think that is the usual way for doing such differentiations across the code.